File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/adapter/api/NativeApi.java
// NativeApi.java

package com.bigboxcode.designpattern.adapter.api;

public class NativeApi implements Api {
    @Override
    public String fetchData() {
        // code to fetch data from native API

        System.out.println("Fetching data from Native API");

        return "Data read from Native Api";
    }

    @Override
    public void sendData(String data) {
        // code to send data to native API

        System.out.println("Sending data to Native API: " + data);
    }
}

File: design-patterns/php/adapter/fileadapter/NativeApi.php
<?php
// NativeApi.php

namespace BigBoxCode\DesignPattern\Adapter\FileAdapter;


class NativeApi implements Api {
    public function fetchData(): string {
        echo "Fetching data from Native API\n";

        return "Data read from Native Api";
    }

    public function sendData(string $data): void {
        echo "Sending data to Native API: " . $data . "\n";
    }
}